src/vellum/lib/test_suite.py

```diff
@@ -0,0 +1,175 @@
+import os
+import time
+from typing import Any, Callable, Union
+from uuid import uuid4
+from ..client import Vellum
+from ..types import (
+    ChatMessage,
+    NamedTestCaseJsonVariableValueRequest,
+    NamedTestCaseNumberVariableValueRequest,
+    NamedTestCaseStringVariableValueRequest,
+    SearchResult,
+    TestSuiteRunExecution,
+    TestSuiteRunMetricOutput,
+    TestSuiteRunState,
+    VellumError,
+    TestSuiteRunExternalExecConfigRequest,
+    TestSuiteRunExternalExecConfigDataRequest,
+    ExternalTestCaseExecution,
+)
+
+
+class VellumTestSuiteResults:
+    def __init__(self, id: str) -> None:
+        self.id = id
+        self.client = Vellum(
+            api_key=os.environ.get("VELLUM_API_KEY"),
+        )
+        self.test_suite_run_state = TestSuiteRunState.QUEUED
+        self.test_suite_run_executions: list[TestSuiteRunExecution] | None = None
```

Comment:
For something like this, I prefer to set to private vars, like `self._test_suite_run_state` and then expose properties for consumers to have read-access of them:

```
@property
def test_suite_run_state(self):
    return self._test_suite_run_state
```

This makes it clear that consumers shouldn't be setting the state of these variables.

Reply:
Note the property can be skipped if we don't think there's a good use-case for consumers to need this info.

Reply:
the original intention was for these to be private